web/client/components/shapefile/SelectShape.jsx

```diff
@@ -49,13 +49,20 @@ class SelectShape extends React.Component {
 
     tryUnzip = (file) => {
         return FileUtils.readZip(file).then((buffer) => {
-            return new JSZip(buffer);
+            var zip = new JSZip();
+            return zip.loadAsync(buffer);
         });
     };
 
-    isZip = (file) => {
+    checkFileType = (file) => {
         return new Promise((resolve, reject) => {
-            if (file.type === 'application/zip' || file.type === 'application/x-zip-compressed') {
+            const ext = FileUtils.recognizeExt(file.name);
+            const type = FileUtils.MIME_LOOKUPS[ext];
```

Comment:
We don't use file.type anymore. I think we should do:
const type = file.type || FileUtils.MIME_LOOKUPS[ext];

Reply:
Mistake